x-pack/plugins/cases/public/components/edit_connector/index.tsx

```diff
@@ -133,13 +133,25 @@ export const EditConnector = React.memo(
       schema,
     });
 
+    // by default save if disabled
+    const [enableSave, setEnableSave] = useState(false);
+
     const { setFieldValue, submit } = form;
 
     const [{ currentConnector, fields, editConnector }, dispatch] = useReducer(
       editConnectorReducer,
       { ...initialState, fields: caseFields }
     );
 
+    // only enable the save button if changes were made to the previous selected
+    // connector or its fields
+    useEffect(() => {
+      const enable =
+        (currentConnector !== undefined && currentConnector?.id !== selectedConnector) ||
+        !shallowEqual(fields, caseFields);
```

Comment:
Why `shallowEqual` and not `deepEqual` ? It is not guaranteed that the field of the connector would be one level deep.

Reply:
I thought they were all one level. I'll change it to deepEqual. thanks for the tip